magpie/api/management/register/register_views.py

```diff
@@ -82,3 +91,21 @@ def leave_discoverable_group_view(request):
     group = ru.get_discoverable_group_by_name(group.group_name, db_session=request.db)
     uu.delete_user_group(user, group, request.db)
     return ax.valid_http(http_success=HTTPOk, detail=s.RegisterGroup_DELETE_OkResponseSchema.description)
+
+
+@s.TemporaryUrlAPI.get(schema=s.TemporaryURL_GET_RequestSchema, tags=[s.RegisterTag],
+                       response_schemas=s.TemporaryURL_GET_responses)
+@view_config(route_name=s.TemporaryUrlAPI.name, request_method="GET", permission=NO_PERMISSION_REQUIRED)
```

Comment:
Should we required the user to be logged in to access the temporary endpoint?
What if someone claim that it never accepts the T&C or that someone resets his password for him?
Could the temporary endpoint redirects to login screen (if no current session) before performing the job?

Reply:
For T&C, the stored `user_id` should be validated against the logged-user so only he can accept the terms.  
So yes definitely requires to be logged in this case. Redirect to login page as need could be a plus if he wasn't logged in.

For reset-password, the whole point is if the user forgot his password. So we cannot ask him to login with it. 
This is why the route is ` NO_PERMISSION_REQUIRED`. No logged in requirement is needed in this case.
Note that this is the "completing the reset" part of the password-reset procedure. 
To trigger the reset itself, either an admin does it or the user during a "forgot my password" kind of login. The user would need to provide the valid username & email in that case, and the generated random token URL received at that email counter-validated with the email account.

Reply:
You're right! All of this make sense.